src/status_im2/navigation/core.cljs

```diff
@@ -126,14 +147,14 @@
         (reset! state/curr-modal true)
         (swap! state/modals conj comp)
         (navigation/show-modal
-         {:component
-          {:name    comp
-           :id      comp
-           :options (merge (options/default-root)
-                           (options/statusbar-and-navbar)
-                           options
-                           (when sheet?
-                             options/sheet-options))}})))))
+         {:stack {:children [{:component
```

Comment:
just wondering why only this change is not enough? why `:navigate-to-within-stack` is needed

Reply:
Because `:navigate-to` calls `dismiss-all-modals`, and also it always calls `navigation/push` with `@state/root` value as componentId, and we need to specify a componentId to trigger the push in the correct stack. We could modify this event to accept componentId as an optional parameter, but I think having separate events is safer.